searx/__init__.py

```diff
@@ -32,52 +32,24 @@
 else:
     logger.info('max_request_timeout=%i second(s)', max_request_timeout)
 
-
-class _brand_namespace:  # pylint: disable=invalid-name
-
-    @classmethod
-    def get_val(cls, group, name, default=''):
-        return settings.get(group, {}).get(name) or default
-
-    @property
-    def SEARX_URL(self):
-        return self.get_val('server', 'base_url')
-
-    @property
-    def CONTACT_URL(self):
-        return self.get_val('general', 'contact_url')
-
-    @property
-    def GIT_URL(self):
-        return self.get_val('brand', 'git_url')
-
-    @property
-    def GIT_BRANCH(self):
-        return self.get_val('brand', 'git_branch')
-
-    @property
-    def ISSUE_URL(self):
-        return self.get_val('brand', 'issue_url')
-
-    @property
-    def NEW_ISSUE_URL(self):
-        return self.get_val('brand', 'new_issue_url')
-
-    @property
-    def DOCS_URL(self):
-        return self.get_val('brand', 'docs_url')
-
-    @property
-    def PUBLIC_INSTANCES(self):
-        return self.get_val('brand', 'public_instances')
-
-    @property
-    def WIKI_URL(self):
-        return self.get_val('brand', 'wiki_url')
-
-    @property
-    def TWITTER_URL(self):
-        return self.get_val('brand', 'twitter_url')
-
-
-brand = _brand_namespace()
+_unset = object()
+
+def get_setting(name, default=_unset):
```

Comment:
Is this function useful?
`settings_default.py`:
* makes sure the value exist.
* sets a default if the value is not set in `settings.yml`.

For example, `get_settings('brand.git_url')` can be replaced by `settings['brand']['git_url']` without issue.

Reply:
This function exists for convenience. Your example is good for values having a default value (in the SCHEMA or in the `settings.yml` file). But what about settings which do not exists by default, those will raise an `KeyError`. With this function you have a default option.

Further I think addressing an option by a doted name is more readable:

    by doted name        : 'brand.git_url'
    by dict's __getitem__: ['brand']['git_url'] 

My be this is only my very personal view on this topic, but I think it is much more comfortable to have a function in templates (or in the `docs/conf.py` or `utils/build_env.py` or ..) which return a defaults whenever an option is unset instead of juggling in the template to avoid an `KeyError` exception.

Reply:
TLDR: The only thing that I would remove if the default value in most (all?) `get_settings` calls : it spreads the default values everywhere.

---

Mandatory values according the AGPL license (the app should stop if there are not defined):
* `git_url`
* `git_branch`

--> no default values. An exception is excepted if the value is undefined.
Side note: reading #36 , perhaps in another PR, the code should get these values from the current git repository (or be overridden by settings.yml value ?). The discussion about this should go into the #36 .

Optional (?) values:
* `issue_url`
* `new_issue_url`
* `docs_url`
* `public_instances`
* `wiki_url`

--> if the value is not specified, the Jinja templates & the documentation should hide the links instead of rendering empty links.

So in theory the schema should be:
```python
    'brand': {
        'git_url': SettingsValue(str),
        'git_branch': SettingsValue(str),
        'issue_url': SettingsValue(str, None),
        'new_issue_url': SettingsValue(str, None),
        'docs_url': SettingsValue(str, None),
        'public_instances': SettingsValue(str, None),
        'wiki_url': SettingsValue(str, None),
    },
```

But in the master branch, the code returns an empty string for undefined values... so actually it doesn't change on this aspect.

---

An alternative solution to `get_settings`:

```python
from typing import Any
from searx import settings


_unset = object()


class Settings:

    __slots__ = 'settings',

    def __init__(self, settings):
        self.settings = settings

    def __getitem__(self, item):
        return self.settings[item]

    def getvalue(self, name: str, default=_unset) -> Any:
        if name in ('settings',):
            return super().__getattribute__(name)
        value = self.settings
        for a in name.split('.'):
            if isinstance(value, dict):
                value = value.get(a, _unset)
            else:
                value = _unset

            if value is _unset:
                if default is _unset:
                    raise KeyError(name)
                value = default
                break

        return value

    def __getattribute__(self, name: str) -> Any:
        """Optional hack: allow to write settings.brand.git_url 
        in a similar way to threading.local
        no default value.

        Might be useful later if the settings.yml becomes a class similar to the Preferences class.
        """
        if name in ('settings', '__slots__', 'getvalue'):
            return super().__getattribute__(name)
        value = self.settings[name]
        if isinstance(value, dict):
            # it would be better to create the Settings instance once, instead at each call.            
            return Settings(value)
        return value

s = Settings(settings)
print(s['brand']['git_url'])  # s still behave as a dict

print(s.getvalue('brand.git_url'))  # getsettings becomes a method. To avoid confusion it is named getvalue not get.
print(s.getvalue('brand.issue_url', 'https://github.com'))  # with a default value

print(s.brand.git_url)  # s has attribute (optional hack)
print(s.engines[0]['engine'])  # can't write s.egines[0].engine
```